File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/NightMessages/Blackmailed.java
package townofsalemcalculator.Conditions.Concrete.NightMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Blackmailer;
import townofsalemcalculator.Conditions.RoleGroup.SingleRoleGroup;

/**
 * The Concrete Condition that you were blackmailed
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-28
 */
public class Blackmailed extends NightMessage {
    public Blackmailed() {
        super(new SingleRoleGroup(Blackmailer));
    }
    
    @Override
    public String getCondition() {
        return "Someone threatened to reveal your secrets. You were blackmailed";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"BMed", "Blackmailed", "Threatened", "Reveal", "Secrets", "Your"});
        keyWords.addAll(Blackmailer.getKeyWords());
        return keyWords;
    }
}

File: src/Conditions/Concrete/Effects/Blackmailed.py
from typing import FrozenSet
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition

class Blackmailed(ExistenceCondition):
    """ The Blackmailed condition is used when at then end of the night you got the message that you are blackmailed,
    i.e. you received the message: 'Someone threatened to reveal your secrets. You were blackmailed' """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.BLACKMAILER}), amnesiacRemembered)